flict/flictlib/arbiter.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: 2022 Henrik Sandklef
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+import sys
+
+from flict.flictlib.lic_comp import LicenseCompatibilty
+from flict.flictlib.project.reader import Project
+from flict.flictlib.utils import meta_information
+from flict.flictlib.utils import timestamp
+
+
+class Arbiter:
+    """Arbiter is a class to verify compatibility"""
+
+    def __init__(self, license_db = None, licenses_preferences = None, denied_licenses=None, alias_file=None):
+        """Initializes Arbiter objects
+             Parameters:
+                 license_db: license database to use instead of builtin
+                 licenses_preferences: license preferences to use instead of builtin
+                 denied_licenses: licenses that cannot be used
+        """
+        self.license_compatibility = LicenseCompatibilty(license_db=license_db, licenses_preferences=licenses_preferences, denied_licenses=denied_licenses, alias_file=alias_file)
+
+    def supported_licenses(self):
+        """Returns the supported licenses"""
+        return self.license_compatibility.supported_licenses()
+
+    def verify_package(self, package, licenses):
+        """Verifies a package's license to a list of outbounds and returns the
+        compatibility between the liceses.
+             Parameters:
+                 package: the package (with its license) to check for compatibility
+                 licenses: the licenses to check the package's license against
+        """
+        logging.debug(f"* verify package {package['name']} (\"{package['license']}\"")
+
+        logging.debug(f"   * All licenses: {licenses}")
+        inbound_license = package['license']
+        logging.debug(f"   * Inbound license:  {inbound_license}")
+
+        compats = []
+        for outbound_license in licenses:
+            compat = self.inbounds_outbound_check(outbound_license, [inbound_license])
+            compats.append(compat)
+
+        return compats
```

Comment:
`return [self.inbounds_outbound_check(outbound_license, [inbound_license]) for outbound_license in licenses]`